applications/luci-app-p910nd/htdocs/luci-static/resources/view/p910nd.js

```diff
@@ -0,0 +1,268 @@
+		}
+		o.rmempty = true;
+
+		o = s.option(form.ListValue, 'port', _('Port'),
+			_('Local TCP listen port for this printer.') + '<br />' +
+			_('Be aware: %s also listens on port 9100.').format('<code>node_exporter</code>'));
+		for (var i = 0; i < 10; i++) {
+			o.value(i, 9100+i);
+		}
+		o.rmempty = true;
+
+		o = s.option(form.Value, 'bind', _('Listen IP'),
+			_('Listen on a specific IP.'));
+		o.datatype = 'ipaddr("nomask")';
+		o.placeholder = _('any');
+		var net_devices = data[0];
+		net_devices.forEach(net_dev => {
+			net_dev.getIPAddrs().forEach(addr => o.value(addr.split('/')[0], E([], [addr.split('/')[0], ' (', E('strong', {}, net_dev.getName()), ')'])));
+			net_dev.getIP6Addrs().forEach(addr => o.value(addr.split('/')[0], E([], [addr.split('/')[0], ' (', E('strong', {}, net_dev.getName()), ')'])));
+		});
+
+		o = s.option(form.Flag, 'bidirectional',
+			_('Bidirectional mode'),
+			_('Whether this print port is bi-directional.')  + '<br />' +
+			_('Note: USB hotplug correctly detects this.'));
+		o.modalonly = true;
+
+		o = s.option(form.Flag, 'runas_root', _('Run as root'),
+			_('Overrides default of %s.').format('<code>user p910nd, group lp</code>'));
+		o.modalonly = true;
+
+		o = s.option(form.Flag, 'mdns', 'mDNS',
+			_('Whether to advertise this printer via %s.', 'mDNS/Bonjour/ZeroConf').format('mDNS/Bonjour/ZeroConf') + '<br />' +
+			_('Note: %s only advertises one (the first) printer on this host.', 'mDNS').format('mDNS') + '<br />' +
+			_('Note: USB hotplug attempts to provide some of the values below.'));
+
+		/* See https://developer.apple.com/bonjour/printing-specification/ ( -> bonjourprinting-1.2.1.pdf ) for
+		 * a description of the Bonjour fields a la Apple.
+		 */
+		o = s.option(form.Value, 'mdns_ty', 'mdns_ty',
+			_('The %s type element.').format('mDNS') + '<br />' +
+			_('User readable description of maker and model.'));
+		o.placeholder = 'HP Color LaserJet CP2025dn';
+		o.rmempty = true;
+		o.optional = true;
+		o.depends({mdns: '1'});
+
+		o = s.option(form.Value, 'mdns_note', 'mdns_note',
+			_('Serves as Location in Apple standards.'));
+		o.placeholder = _('By the router');
+		o.rmempty = true;
+		o.optional = true;
+		o.modalonly = true;
+		o.depends({mdns: '1'});
+
+		o = s.option(form.Value, 'mdns_product', 'mdns_product',
+			_('The %s value.').format('<code>IEEE1284 Product</code>') + '<br />' +
+			_('Note: must be %s.').format('<code>(</code>' + _('enclosed within parentheses') + '<code>)</code>'));
+		o.placeholder = '(Color LaserJet CP2025dn)';
+		o.rmempty = true;
+		o.optional = true;
+		o.modalonly = true;
+		o.depends({mdns: '1'});
+		o.write = function(section_id, value) {
+			if (value){
+				if (!value.startsWith('('))
+					value = '(' + value;
+				if (!value.endsWith(')'))
+					value = value + ')';
+				if (value === '()')
+					return;
+				uci.set(pkg.Name, section_id, 'mdns_product', value);
+			}
+		}
+
+		const string_to_array = function(string, uppercase=false) {
```

Comment:
Can the function be replaced with a split only?
The uppercase always false, the input strings are constants that we can  sort manually

Reply:
It would be naïve at best.